### PR TITLE
Added commands. working undo/redo

### PR DESCRIPTION
**Pull request by pgarneau**

### CODE REVIEW

File: src/main/java/apaintus/controllers/CanvasController.java [hunk 1]
```diff
@@ -56,71 +60,62 @@ public void injectParentController(Controller controller) {
         this.updateService = new UpdateService(this.attributeController, this.toolBarController);
 
         snapGrid = new SnapGrid(toolBarController.getGridSpacing(),canvas.getWidth(),canvas.getHeight(),toolBarController.getStrokeSize(),false);
+        invoker = this.controller.getInvoker();
     }
 
     @Override
     public void initialize() {
+
```
Comment (pspino) [inline src/main/java/apaintus/controllers/CanvasController.java]: i guess there is a added space here or git diff is doing some weird shit

File: src/main/java/apaintus/controllers/CanvasController.java [hunk 2]
```diff
@@ -56,71 +60,62 @@ public void injectParentController(Controller controller) {
...
             }
         });
 
         canvas.setOnMouseReleased(event -> {
+            deselectCommand.undo();
+
             if (event.getX() == lastMouseClickPosition.getX() && event.getY() == lastMouseClickPosition.getY()) {
-                return;
-            }
+                if (activeTool == ActiveTool.SELECT) {
```
Comment (pspino) [inline src/main/java/apaintus/controllers/CanvasController.java:85]: why do we do something when we click and we don't move. I don't get the logic behind. previously we were not doing anything, why the change then ?

File: src/main/java/apaintus/controllers/CanvasController.java [hunk 3]
```diff
@@ -269,8 +281,8 @@ public ColorChangeListener(Attribute attribute) {
         @Override
         public void changed(ObservableValue<? extends Color> observableValue, Color oldValue, Color newValue) {
             if(activeShape != null && activeShape.isSelected() && drawnShapes.contains(activeShape)) {
-                activeShape.update(updateService.getAttributes());
-                redrawCanvas();
+                System.out.println("COLOR CHANGED");
```
Comment (pspino) [inline src/main/java/apaintus/controllers/CanvasController.java]: remove sout pls

File: src/main/java/apaintus/controllers/CanvasController.java [hunk 4]
```diff
@@ -287,16 +299,16 @@ public void changed(ObservableValue<? extends Double> observableValue, Double ol
             if(activeShape != null && activeShape.isSelected() && drawnShapes.contains(activeShape)) {
                 if(activeShape.getShapeType() == ShapeType.SELECTION_BOX) {
                     ((SelectionBox) activeShape).update(attribute, newValue - oldValue);
+                    redrawCanvas();
                 } else {
-                    activeShape.update(updateService.getAttributes());
+                    System.out.println("SPINNER CHANGED");
```
Comment (pspino) [inline src/main/java/apaintus/controllers/CanvasController.java]: remove sout pls

File: src/main/java/apaintus/controllers/CanvasController.java (hunk 2 above)
Comment (pgarneau) [reply src/main/java/apaintus/controllers/CanvasController.java:85]: Quand tu click et t'as pas bouge, il check pour une selection / deselection

**Review by boscoa — APPROVED** [review]
